docker/docker-compose.dev.yml

```diff
@@ -87,6 +89,15 @@ services:
     depends_on:
       - immich-server
 
+  typesense:
+    container_name: immich_typesense
+    image: typesense/typesense:0.24.0
+    environment:
+      - TYPESENSE_API_KEY=${TYPESENSE_API_KEY}
```

Comment:
Do we need to use the key for a self-hosted instance?

Reply:
Ah I saw the default self-hosted key is `xyz`, should we put this somewhere as the default?

Reply:
Probably. I haven't focused too much on that. I've only added the container to the dev compose lol. It works and we can clean up the configuration and do something sensible. We could just require an API key or throw an error too.

Reply:
I would say we treat it similar to the postgres password, with a default in the example.env

Reply:
We can do that, this is unrelated though as it's just a type error because process.env is of type `string | undefined` so we can probably just cast it.

Reply:
That and it's required in `app.config.ts`, so the app won't start without it. We can do the same for this.